models/service_test.go

```diff
@@ -0,0 +1,205 @@
+				models.Address{Kind: "Pod", Name: "recommendation-v2", IP: "172.17.0.8"},
+			},
+			Ports: models.Ports{
+				models.Port{Name: "http", Protocol: "TCP", Port: 3001},
+				models.Port{Name: "http", Protocol: "TCP", Port: 3000},
+			}}})
+	assert.Equal(t, service.Pods, models.Pods{
+		models.Pod{
+			Name: "Pod1",
+			Labels: map[string]string{
+				"label1": "labelName1Pod1",
+				"label2": "labelName2Pod1"}},
+		models.Pod{
+			Name: "Pod2",
+			Labels: map[string]string{
+				"label1": "labelName1Pod2",
+				"label2": "labelName2Pod2"}}})
+
+	// Istio Details
+	assert.Equal(t, service.RouteRules, models.RouteRules{
+		models.RouteRule{
+			Destination: "destination1",
+			Precedence:  "precedence1",
+			Route:       "route1"},
+		models.RouteRule{
+			Destination: "destination2",
+			Precedence:  "precedence2",
+			Route:       "route2"}})
+
+	// Prometheus Client
+	assert.Equal(t, service.Dependencies, map[string][]string{
+		"v1": []string{"unknown", "/products", "/reviews"},
+		"v2": []string{"/catalog", "/shares"}})
+}
+
+func fakeServiceDetails() *kubernetes.ServiceDetails {
+	service := &v1.Service{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      "Name",
+			Namespace: "Namespace",
+			Labels: map[string]string{
+				"label1": "labelName1",
+				"label2": "labelName2"}},
+		Spec: v1.ServiceSpec{
+			ClusterIP: "fromservice",
+			Type:      "ClusterIP",
+			Ports: []v1.ServicePort{
+				v1.ServicePort{
+					Name:     "http",
+					Protocol: "TCP",
+					Port:     3001},
+				v1.ServicePort{
+					Name:     "http",
+					Protocol: "TCP",
+					Port:     3000}}}}
+
+	endpoints := &v1.Endpoints{
+		Subsets: []v1.EndpointSubset{
+			v1.EndpointSubset{
+				Addresses: []v1.EndpointAddress{
+					v1.EndpointAddress{
+						IP: "172.17.0.9",
+						TargetRef: &v1.ObjectReference{
+							Kind: "Pod",
+							Name: "recommendation-v1"}},
+					v1.EndpointAddress{
+						IP: "172.17.0.8",
+						TargetRef: &v1.ObjectReference{
+							Kind: "Pod",
+							Name: "recommendation-v2"}},
+				},
+				Ports: []v1.EndpointPort{
+					v1.EndpointPort{Name: "http", Protocol: "TCP", Port: 3001},
+					v1.EndpointPort{Name: "http", Protocol: "TCP", Port: 3000},
+				}}}}
+
+	pods := []*v1.Pod{
+		&v1.Pod{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name:      "Pod1",
+				Namespace: "Namespace1",
+				Labels: map[string]string{
+					"label1": "labelName1Pod1",
+					"label2": "labelName2Pod1"}}},
+		&v1.Pod{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name:      "Pod2",
+				Namespace: "Namespace2",
+				Labels: map[string]string{
+					"label1": "labelName1Pod2",
+					"label2": "labelName2Pod2"}}}}
+
+	return &kubernetes.ServiceDetails{service, endpoints, pods}
+}
+
+func fakeIstioDetails() *kubernetes.IstioDetails {
+	routes := []*kubernetes.RouteRule{
+		&kubernetes.RouteRule{
+			Spec: map[string]interface{}{
+				"destination": "destination1",
+				"precedence":  "precedence1",
```

Comment:
it would be nicer to fill this with a more meaningful info. it won't change the test outcome, but will be more clear to whoever reads the test.